pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -190,26 +191,34 @@ public BrokerResponseNative handleRequest(JsonNode request, @Nullable RequesterI
     if (sql == null) {
       throw new BadQueryRequestException("Failed to find 'sql' in the request: " + request);
     }
-    return handleRequest(requestId, sql.asText(), request, requesterIdentity, requestContext);
+    String query = sql.asText();
+    requestContext.setQuery(query);
+    return handleRequest(requestId, query, sqlNodeAndOptions, compilationStartTimeNs, request, requesterIdentity,
+        requestContext);
   }
 
-  private BrokerResponseNative handleRequest(long requestId, String query, JsonNode request,
+  private BrokerResponseNative handleRequest(long requestId, String query,
+      @Nullable SqlNodeAndOptions sqlNodeAndOptions, long compilationStartTimeNs, JsonNode request,
```

Comment:
passing in compilationStartTimeNs is kinda weird here. let's just break this into parser time and still keep the `compilationStartTimeNs`

Reply:
Wanted to keep the existing behavior. Added `compilationStartTimeNs` into `SqlNodeAndOptions` to keep the interface clean